packages/platforms/delta/arm64/tn48m/src/arm64_delta_tn48m/module/src/psui.c

```diff
@@ -410,9 +419,20 @@ onlp_psui_info_get(onlp_oid_t id, onlp_psu_info_t* info)
                 info->status |= ONLP_PSU_STATUS_FAILED;
         }
 
-        // TODO : auto detect AC / DC type
-        // Just do a guess
+        // Auto detect AC / DC type
+        // The PSU capability is decided by Vin value
         info->caps |= _psu_vin_type_guess(info->mvin);
+    } else {
+        // The PSU capability is decided by configuration
+        if (psu->type == PLAT_PSU_TYPE_AC) {
+            info->caps |= ONLP_PSU_CAPS_AC;
+        } else if (psu->type == PLAT_PSU_TYPE_DC12) {
+            info->caps |= ONLP_PSU_CAPS_DC12;
+        } else if (psu->type == PLAT_PSU_TYPE_DC48) {
+            info->caps |= ONLP_PSU_CAPS_DC48;
+        } else {
+            info->status |= ONLP_PSU_STATUS_FAILED;
+        }
```

Comment:
A switch statement could be used too, but the necessary `break;` statements would make it longer in the end. So nothing to be done.